packages/backend/src/open_payments/payment/incoming/service.ts

```diff
@@ -247,14 +260,21 @@ async function handleDeactivated(
 async function getPaymentPointerPage(
   deps: ServiceDependencies,
   paymentPointerId: string,
-  pagination?: Pagination
+  pagination?: Pagination,
+  clientId?: string
 ): Promise<IncomingPayment[]> {
-  const page = await IncomingPayment.query(deps.knex)
+  let page: IncomingPayment[]
+  page = await IncomingPayment.query(deps.knex)
     .getPage(pagination)
     .where({
       paymentPointerId
     })
-    .withGraphFetched('asset')
+    .withGraphFetched('[asset, grant]')
+  if (clientId) {
+    page = page.filter(
+      (payment: IncomingPayment) => payment.grant.clientId === clientId
```

Comment:
This might need to happen within `getPage` in order to adhere to the `pagination` parameters

Reply:
Good point. Didn't think about that. It was incredible hard to change that to `withGraphJoined` so I gave up, but I'll give it another go.